rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/TestRestDocsSpecBase.java

```diff
@@ -64,4 +81,23 @@ public void setupRestDocs() {
                 .alwaysDo(this.documentationHandler)
                 .build();
     }
+
+    public void testAttachmentUpload(String url , String id) throws Exception {
+        String attachment = "{ \"filename\":\"spring-core-4.3.4.RELEASE.jar\", \"attachmentContentId\":\"2\"}";
+		/*
+		 * TODO Suggestion to use better library in future, instead of MockMultipartFile to have better document generation feature.
+		 * below logic is used to generate the correct restapi end point documentation
+		 */
+        MockMultipartFile jsonFile = new MockMultipartFile("attachment","","application/json",new ByteArrayInputStream(attachment.getBytes()));
+		String accessToken = TestHelper.getAccessToken(mockMvc, testUserId, testUserPassword);
+		MockHttpServletRequestBuilder builder = MockMvcRequestBuilders
+				.fileUpload(url + id + "/attachments")
+				.file("file", "@/spring-core-4.3.4.RELEASE.jar".getBytes())
+				.file(jsonFile)
+				.contentType(MediaType.MULTIPART_FORM_DATA)
+				.header("Authorization", "Bearer " + accessToken);
+		this.mockMvc.perform(builder)
+					.andExpect(status().isOk())
+					.andDo(this.documentationHandler.document());
```

Comment:
I think one of the main parts of the documentation is missing here, where you would describe the request and response fields. Or is there a special reason for not describing them?

Reply:
I did not take a look at the api method itself, but I am wondering also if the api client should really give a attachment content id?

Reply:
I found few issues regarding this in spring community ,specifically for MultiPartFile. It does not have proper documentation support. So I am not sure if we can generate the above.Let me try if we can do this.

Reply:
https://github.com/spring-projects/spring-restdocs/issues/104
https://github.com/spring-projects/spring-restdocs/issues/48

Reply:
For this request using MultiPartFile , it is not supporting to add request fields . So as discussed, I have added the request fields in in adoc file.